client_plugin/drivers/shared/etcdops.go

```diff
@@ -280,3 +292,104 @@ func addrToEtcdClient(addr string) (*etcdClient.Client, error) {
 
 	return cli, nil
 }
+
+// writeVolMetadata - Update or Create volume metadata in KV store
+func (d *VolumeDriver) writeVolMetadata(entries []kvPair) error {
+
+	var ops []etcdClient.Op
+	var msg string
+	var err error
+
+	// Create a client to talk to etcd
+	etcdAPI := d.createEtcdClient()
+	defer etcdAPI.Close()
+
+	// ops hold multiple operations that will be done to etcd
+	// in a single revision
+	for _, elem := range entries {
+		ops = append(ops, etcdClient.OpPut(elem.key, elem.value))
+	}
+
+	// Lets write the metadata in a single transaction
+	// Use a transaction if more than one entries are to be written
+	if len(entries) > 1 {
+		_, err = etcdAPI.Txn(context.TODO()).Then(ops...).Commit()
+	} else {
+		_, err = etcdAPI.Do(context.TODO(), ops[0])
+	}
+
+	if err != nil {
+		msg = fmt.Sprintf("Failed to write metadata. Reason: %v", err)
+		log.Warningf(msg)
+		return errors.New(msg)
+	}
+	return nil
+}
+
+// readVolMetadata - Read volume metadata in KV store
+func (d *VolumeDriver) readVolMetadata(keys []string) ([]kvPair, error) {
+	var entries []kvPair
+	var msg string
+
+	// Create a client to talk to etcd
+	etcdAPI := d.createEtcdClient()
+	defer etcdAPI.Close()
+
+	// Read does not require transactions
+	// Just read all the keys required
+	for _, k := range keys {
+		elem := kvPair{key: k, value: ""}
+		ctx, cancel := context.WithTimeout(context.Background(),
+			requestTimeout)
+		resp, err := etcdAPI.Get(ctx, k)
```

Comment:
From what I understand, wrapping GET operations inside a transaction will prevent modifying one of the keys in the middle.
According to https://coreos.com/blog/transactional-memory-with-etcd3.html
```
getresp, err := etcd.Txn(ctx.TODO()).Then(OpGet(from), OpGet(to)).Commit()
    if err != nil {
         return false, err
    }
```
indicate:

>  All Get requests are wrapped in a transaction so interleaved writes cannot update from while fetching to.